app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/returns/GlarCalculatorEligibilityUtil.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsfrontend.services.returns
+
+import cats.data.EitherT
+import cats.instances.future._
+import cats.instances.list._
+import cats.syntax.eq._
+import cats.syntax.traverse._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.Error
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.JourneyStatus.FillingOutReturn
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.AssetType.{NonResidential, Residential}
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.CompleteReturn.CompleteSingleDisposalReturn
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.DraftSingleDisposalReturn
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.IndividualUserType.Self
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.ReliefDetailsAnswers.CompleteReliefDetailsAnswers
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.returns.SingleDisposalTriageAnswers.CompleteSingleDisposalTriageAnswers
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+
+import scala.concurrent.{ExecutionContext, Future}
+@ImplementedBy(classOf[GlarCalculatorEligibilityUtilImpl])
+trait GlarCalculatorEligibilityUtil {
+
+  def isEligibleForFurtherReturnOrAmendCalculation(fillingOutReturn: FillingOutReturn)(implicit
+    hc: HeaderCarrier
+  ): EitherT[Future, Error, Boolean]
+
+}
+
+@Singleton
+class GlarCalculatorEligibilityUtilImpl @Inject() (
+  returnsService: ReturnsService,
+  servicesConfig: ServicesConfig
+)(implicit ec: ExecutionContext)
+    extends GlarCalculatorEligibilityUtil {
+
+  val amendAndFurtherReturnCalculationsEnabled = servicesConfig.getBoolean("glar-calculator.enabled")
+
+  def isEligibleForFurtherReturnOrAmendCalculation(
+    fillingOutReturn: FillingOutReturn
+  )(implicit headerCarrier: HeaderCarrier): EitherT[Future, Error, Boolean] = {
+
+    @SuppressWarnings(Array("org.wartremover.warts.Any"))
```

Comment:
there shouldn't be an instance of `Any` here - 99% of the time, if there is an `Any` then something's gone wrong with the types which needs to be solved